lib/plugins/platform/platform.js

```diff
@@ -0,0 +1,174 @@
+const gql = require('graphql-tag');
+const jwtDecode = require('jwt-decode');
+const BbPromise = require('bluebird');
+const fsExtra = require('../../utils/fs/fse');
+const fetch = require('node-fetch');
+const configUtils = require('../../utils/config');
+const functionInfoUtils = require('../../utils/functionInfoUtils');
+const createApolloClient = require('../../utils/createApolloClient');
+
+// NOTE Needed for apollo to work
+global.fetch = fetch;
+
+const config = {
+  PLATFORM_FRONTEND_BASE_URL: 'https://platform.serverless.com/',
+  GRAPHQL_ENDPOINT_URL: 'https://graphql.serverless.com/graphql',
+};
+
+function addReadme(attributes, readmePath) {
+  if (fs.existsSync(readmePath)) {
+    const readmeContent = fsExtra.readFileSync(readmePath).toString('utf8');
+    // eslint-disable-next-line no-param-reassign
+    attributes.readme = readmeContent;
+  }
+  return attributes;
+}
+
+function fetchEndpoint(provider, stage, region) {
+  return provider
+    .request(
+      'CloudFormation',
+      'describeStacks',
+      { StackName: provider.naming.getStackName(stage) },
+      stage,
+      region
+    )
+    .then(result => {
+      let endpoint = null;
+
+      if (result) {
+        result.Stacks[0].Outputs
+          .filter(x => x.OutputKey.match(provider.naming.getServiceEndpointRegex()))
+          .forEach(x => {
+            endpoint = x.OutputValue;
+          });
+      }
+
+      return endpoint;
+    });
+}
+
+class Platform {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+    // NOTE for the time being we only track services published to AWS
+    if (this.provider) {
+      this.hooks = {
+        'after:deploy:deploy': this.publishService.bind(this),
+      };
+    }
+  }
+
+  publishServiceRequest(service, client) {
+    return client
+      .mutate({
+        mutation: gql`
+        mutation publishService($service: ServicePublishInputType!) {
+          publishService(service: $service) {
+            name
+          }
+        }
+      `,
+        variables: { service },
+      })
+      .then(response => response.data);
+  }
+
+  getAuthToken() {
+    const userConfig = configUtils.getConfig();
+    const currentId = userConfig.userId;
+    const globalConfig = configUtils.getGlobalConfig();
+    if (globalConfig.users && globalConfig.users[currentId] && globalConfig.users[currentId].auth) {
+      return globalConfig.users[currentId].auth.id_token;
+    }
+    return null;
+  }
+
+  publishService() {
+    const authToken = this.getAuthToken();
+    if (!authToken) {
+      // NOTE publishService is an opt-in feature and no warning is needed
+      return BbPromise.resolve();
+    }
+    this.serverless.cli.log('Publish service to Serverless Platform...');
+
+    const clientWithAuth = createApolloClient(config.GRAPHQL_ENDPOINT_URL, authToken);
+
+    const region = this.serverless.service.provider.region;
```

Comment:
yeah, thats why I thought it should be moved to the aws directory, to make use of those common functionality